### ISSUE TITLE
nDroid Project Seems to not be Compatible with TI Nspire CX II CAS

### ISSUE THREAD

**Issue by logantischler0** [ndless-nspire/Ndless#282]
I cannot figure out how to get nDroid (https://github.com/nDroidProject) on my TI Nspire CX II CAS. I looked at this tutorial: https://forum.xda-developers.com/t/port-calculator-android-on-the-ti-nspire-cx.3143757/ However, "Step 2: Download files", gives you links that don't work. It says, "404 Not Found", on all of them.

Any suggestions?

Thanks in advance.

**Comment by Facts-and-Logic** [comment]
Using the Wayback Machine will let you download the files. Here are the fixed links:

Boot.img.tns: [Link](https://web.archive.org/web/20201201080043/https://get.xda-developers.com/dl/1/0/5/1/2/boot.img.tns?key=BxEL8BY72qOlOdbg0RROCA&ts=1606809703)
System.tar.gz: [Link](https://web.archive.org/web/20201201080052/https://get.xda-developers.com/dl/1/0/5/1/2/System.tar.gz?key=EESglDET-SduaYR-cCIvmA&ts=1606809711)
nspire-cx.dtb.tns: [Link](https://web.archive.org/web/20201201080048/https://get.xda-developers.com/dl/1/0/5/1/2/nspire-cx.dtb.tns?key=SeUZtiZjqN-rt2NlJwEQgQ&ts=1606809708)
nDroidLoader.tns: [Link](https://web.archive.org/web/20201201080046/https://get.xda-developers.com/dl/1/0/5/1/2/ndroidloader.tns?key=4mveJbwVNZMsmcaftG2QQg&ts=1606809706)
startndroid.ll2.tns: [Link](https://web.archive.org/web/20201201080049/https://get.xda-developers.com/dl/1/0/5/1/2/startndroid.ll2.tns?key=eGgwWHg0EV7GDIXoB0xqRg&ts=1606809709)

**Comment by logantischler0** [comment]
I would never have guessed that archive.org would work with this! Thank you so much!

**Comment by logantischler0** [comment]
I've encountered another issue.

Connected to the calculator, there's a Mini-USB to USB (Female) adapter. There's a USB hub connected to that, which has 3 USB ports, as well as an Ethernet port. In one of the USB ports, I have connected the USB drive that has been formatted according to https://forum.xda-developers.com/t/port-calculator-android-on-the-ti-nspire-cx.3143757/. (System.tar.gz has been extracted onto the first partition of the USB drive, as the article instructs.) Connected to another one of the USB hub's USB ports is a USB keyboard.

About 2 minutes after running startndroid.ll2.tns, the calculator hardlocks after displaying the line, "random: nonblocking pool is initialized". I've tried to get beyond this point on both my CX II CAS as well as my CX CAS, but I've had no success. (The CX II CAS has glitched graphics, however, so it is preferable to use the CX CAS.)

Any suggestions?

Thanks in advance!

**Comment by adriweb** [comment]
Yeah for Linux on the CX II at least, it's sufficiently different than kernel patches had to be done (see https://tiplanet.org/forum/viewtopic.php?f=20&t=24785) so it doesn't surprise me that nDroid doesn't work as is either.

**Comment by Vogtinator** [comment]
That basically means the kernel is stuck either trying to find the rootfs or already in userspace. You can try raising the loglevel a bit to get more info, but I'm not really familiar with nDroid.

**Comment by SuperX-dev** [comment]
> I've encountered another issue.
> 
> Connected to the calculator, there's a Mini-USB to USB (Female) adapter. There's a USB hub connected to that, which has 3 USB ports, as well as an Ethernet port. In one of the USB ports, I have connected the USB drive that has been formatted according to https://forum.xda-developers.com/t/port-calculator-android-on-the-ti-nspire-cx.3143757/. (System.tar.gz has been extracted onto the first partition of the USB drive, as the article instructs.) Connected to another one of the USB hub's USB ports is a USB keyboard.
> 
> About 2 minutes after running startndroid.ll2.tns, the calculator hardlocks after displaying the line, "random: nonblocking pool is initialized". I've tried to get beyond this point on both my CX II CAS as well as my CX CAS, but I've had no success. (The CX II CAS has glitched graphics, however, so it is preferable to use the CX CAS.)
> 
> Any suggestions?
> 
> Thanks in advance!

I have here a Ti-Nspire CX HW-Y and it seems that nDroid has also problems with my calculator

So even if you get it to work in a Kernel side I think the LCD will also be a problem. I don´t think I´ve made something wrong in the process, but the fact that in the original XDA Developers post it targets the OS 3.1 can mean that only the old LCD is supported.

I´m not really sure if this is true, but I´m posting it so you to know.

Here´s a photo of my calculator after 15minutes of trying to open nDroid

![IMG_20210804_085456](https://user-images.githubusercontent.com/74361788/128150671-62cb7fda-6c3a-414c-a486-ddf676f3fee2.jpg)

Maybe it has something to do with lcd blit support? 

Note:Tried in OS 4.4 and 4.5

**Comment by Vogtinator** [comment]
That's how it looks like if the LCD isn't initialized correctly, but the backlight is on. If it's just about the different lcd panel, it would have garbled content instead. That would be fixed by using the LCD panel config from the CX II DTB, but this appears to be a different issue.

**Comment by sasdallas** [comment]
While a bit unconventional @satyamedh has gotten Android 2.3 Gingerbread to launch on the Nspire CX II.
https://github.com/satyamedh/nDroid_linux_kernel
